cmd/nerdctl/run_network.go

```diff
@@ -110,75 +111,88 @@ func withCustomHosts(src string) func(context.Context, oci.Client, *containers.C
 	}
 }
 
-func generateNetOpts(cmd *cobra.Command, dataStore, stateDir, ns, id string) ([]oci.SpecOpts, []string, string, []gocni.PortMapping, error) {
+func generateNetOpts(cmd *cobra.Command, dataStore, stateDir, ns, id string) ([]oci.SpecOpts, []string, string, []gocni.PortMapping, string, error) {
 	opts := []oci.SpecOpts{}
 	portSlice, err := cmd.Flags().GetStringSlice("publish")
 	if err != nil {
-		return nil, nil, "", nil, err
+		return nil, nil, "", nil, "", err
 	}
 	ipAddress, err := cmd.Flags().GetString("ip")
 	if err != nil {
-		return nil, nil, "", nil, err
+		return nil, nil, "", nil, "", err
 	}
 	netSlice, err := getNetworkSlice(cmd)
 	if err != nil {
-		return nil, nil, "", nil, err
+		return nil, nil, "", nil, "", err
 	}
 
 	if (len(netSlice) == 0) && (ipAddress != "") {
 		logrus.Warnf("You have assign an IP address %s but no network, So we will use the default network", ipAddress)
 	}
 
+	macAddress, err := getMACAddress(cmd, netSlice)
+	if err != nil {
+		return nil, nil, "", nil, "", err
+	}
+
 	ports := make([]gocni.PortMapping, 0)
 	netType, err := nettype.Detect(netSlice)
 	if err != nil {
-		return nil, nil, "", nil, err
+		return nil, nil, "", nil, "", err
 	}
 
 	switch netType {
 	case nettype.None:
 		// NOP
+		// Docker compatible: if macAddress is specified, set MAC address shall
+		// not work but run command will success
 	case nettype.Host:
+		if macAddress != "" {
+			return nil, nil, "", nil, "", errors.New("conflicting options: mac-address and the network mode")
+		}
 		opts = append(opts, oci.WithHostNamespace(specs.NetworkNamespace), oci.WithHostHostsFile, oci.WithHostResolvconf)
 	case nettype.CNI:
```

Comment:
Again for my understanding what is the difference between nettype.CNI and nettype.container

Reply:
`nettype.container` is the network shared with other containers, other containers can be CNI network, host network

Reply:
> Support for --network=bridge.
> 
> Partial support for --network=<network> (when actual network types are one of bridge and macvlan).

Just as I've mentioned at the beginning of this conversation, only `bridge`, `macvlan`, and `<network>` which is actually one of them under the hook are supported. So `nettype.Container` is picked up for notifying users this is not supported.